### PR TITLE
feat(dashboard,types,js-sdk,payment): ability to refund payment in order page

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds ability to refund payment in order page

**observations:**

1. when creating a single payment, we validate that the refund is not higher than the payment amount. This however does not account for sending multiple refunds. You can send multiple refunds that gross greater than the payment amount. 
2. In the designs, we have refund reasons, note and send notification. None of these have been implemented in the backend yet. 
3. Currency input breaks on changing input. Will take a look at this separately. In this PR, its a simple number input. 

https://github.com/user-attachments/assets/0ba9df14-0beb-4830-9591-9ae135794729

### CODE REVIEW

**Comment by olivermrbl** [comment]
Looks like the refund entry deviates from the design:

**Video**
CleanShot 2024-08-01 at 13 12 08

**Design**
CleanShot 2024-08-01 at 13 12 47

**Comment by riqwan** [comment]
@olivermrbl I've moved the date and fixed the icon. The issue with the badge and the note icon will still persists as I've written in the observations. Those are not done in the backend, i'm unsure if they're on purpose or not. 


Screenshot 2024-08-01 at 17 06 34

File: packages/admin-next/dashboard/src/lib/client/payments.ts [hunk 1]
```diff
@@ -1,5 +1,13 @@
-import { getRequest } from "./common"
+import { HttpTypes } from "@medusajs/types"
 import { PaymentProvidersListRes } from "../../types/api-responses"
+import { getRequest } from "./common"
+
+async function retrievePayment(id: string, query?: Record<string, any>) {
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/lib/client/payments.ts]: suggestion: Should we migrate this to the `js-sdk` while we are at it?

**Review by olivermrbl — APPROVED** [review]
LGTM, are we tracking the observations in the PR description?

File: packages/admin-next/dashboard/src/lib/client/payments.ts (hunk 1 above)
Comment (riqwan) [reply packages/admin-next/dashboard/src/lib/client/payments.ts]: yes def! force of habit 😄

**Pull request merged**